File: ej2.hs
-- Sitio para ejecutarlo https://www.tutorialspoint.com/compile_haskell_online.php

-- Lista de enteros
listaEnteros :: [Int]
listaEnteros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

-- Potencia n
n :: Int
n = 2

-- Calcular la potencia n-ésima de cada elemento en la lista usando lambda
resultado :: [Int]
resultado = map (\x -> x ^ n) listaEnteros

-- Mostrar el resultado
main :: IO ()
main = print resultado

File: ej2.py
# Lista de enteros
lista_enteros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

# Potencia n
n = 3

# Calcular la potencia n-ésima de cada elemento en la lista usando lambda
resultado = list(map(lambda x: x ** n, lista_enteros))

# Mostrar el resultado
print(resultado)
